tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py

```diff
@@ -0,0 +1,238 @@
+#     c = 0.15
+#     res_amplifier = 5.
+#     resolution_func *= 1/(1+gaussian(center=r_0, c=c, a=res_amplifier))
+
+# mesh1d = CircleManifoldMesh(ncells, radius=rmin, degree=2)  # construct a circle mesh
+# mesh = ExtrudedMesh(mesh1d, layers=nlayers, layer_height=(rmax-rmin)*resolution_func/np.sum(resolution_func), extrusion_type='radial')  # extrude into a cylinder
+
+with CheckpointFile("initial_condition_mat_prop/Final_State.h5", mode="r") as f:
+    mesh = f.load_mesh("firedrake_default_extruded")
+# We set the mesh `cartesian` attribute to False, which ensures that
+# the unit vector points radially, in the direction opposite to gravity.
+mesh.cartesian = False
+bottom_id, top_id = "bottom", "top"
+
+V = VectorFunctionSpace(mesh, "CG", 2)  # Velocity function space (vector)
+V_scalar = FunctionSpace(mesh, "CG", 2)  # CG2 Scalar function space
+W = FunctionSpace(mesh, "CG", 1)  # Pressure function space (scalar)
+Q = FunctionSpace(mesh, "DQ", 2)  # Temperature function space (scalar)
+Z = MixedFunctionSpace([V, W])  # Mixed function space.
+
+z = Function(Z)  # A field over the mixed function space Z.
+u, p = split(z)  # Returns symbolic UFL expression for u and p
+z.subfunctions[0].rename("Velocity")
+z.subfunctions[1].rename("Pressure")
+vr = Function(V_scalar, name="Radial_Velocity")  # For diagnostic output
+
+# We next specify the important constants for this problem, and set up the approximation.
+Ra = Constant(5.0e7)  # Rayleigh number
+Di = Constant(0.9492824165791792)  # Dissipation number
+H_int = Constant(9.93)  # Internal heating
+
+X = SpatialCoordinate(mesh)
+r = sqrt(X[0]**2 + X[1]**2)
+rhat = as_vector([X[0]/r, X[1]/r])  # Radial unit vector (in direction opposite to gravity)
+thetahat = as_vector([-X[1]/r, X[0]/r])  # Tangential unit vector.
+
+rhobar = Function(Q, name="CompRefDensity")
+interpolate_1d_profile(function=rhobar, one_d_filename="initial_condition_mat_prop/rhobar.txt")
+rhobar.assign(rhobar / 3200.)
+Tbar = Function(Q, name="CompRefTemperature")
+interpolate_1d_profile(function=Tbar, one_d_filename="initial_condition_mat_prop/Tbar.txt")
+Tbar.assign((Tbar - 1600.) / 3700.)
+alphabar = Function(Q, name="IsobaricThermalExpansivity")
+interpolate_1d_profile(function=alphabar, one_d_filename="initial_condition_mat_prop/alphabar.txt")
+alphabar.assign(alphabar / 4.1773e-05)
+cpbar = Function(Q, name="IsobaricSpecificHeatCapacity")
+interpolate_1d_profile(function=cpbar, one_d_filename="initial_condition_mat_prop/CpSIbar.txt")
+cpbar.assign(cpbar / 1249.7)
+gbar = Function(Q, name="GravitationalAcceleration")
+interpolate_1d_profile(function=gbar, one_d_filename="initial_condition_mat_prop/gbar.txt")
+gbar.assign(gbar / 9.8267)
+
+# We next prepare our viscosity, starting with a radial profile.
+mu_rad = Function(Q, name="Viscosity_Radial")  # Depth dependent component of viscosity
+interpolate_1d_profile(function=mu_rad, one_d_filename="initial_condition_mat_prop/mu2_radial.txt")
```

Comment:
None of these profiles are included in this PR

Reply:
Added.